[Tables]Build & Test failing in nightly runs

Tables nightly test runs are failing with:

> Error message:
> NUnit Adapter 4.3.1.0: Test execution complete
> Data collector 'Blame' message: All tests finished running, Sequence file will not be generated.
> Results File: /mnt/vss/_work/1/s/sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/tests/TestResults/_7ebf9884c00000J_2023-01-19_09_24_39.trx
> 
> Test Run Successful.
> Total tests: 449
>      Passed: 433
>     Skipped: 16
>  Total time: 2.6328 Minutes
> ##[error]The operation was canceled.

For more details check here:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2127512&view=results

@jsquire for notification.

--- a/sdk/tables/Azure.Data.Tables/samples/Sample4QueryEntities.md
+++ b/sdk/tables/Azure.Data.Tables/samples/Sample4QueryEntities.md
@@ -102,8 +102,8 @@ Below is a simple example that illustrates how the token can be fetched and appl
 
 ```C# Snippet:TablesSample4QueryPagination
 string continuationToken = null;
-bool hadResults = true;
-while (hadResults)
+bool moreResultsAvailable = true;
+while (moreResultsAvailable)
 {
     Page<TableEntity> page = tableClient
         .Query<TableEntity>()
@@ -118,7 +118,7 @@ while (hadResults)
     continuationToken = page.ContinuationToken;
 
     IReadOnlyList<TableEntity> pageResults = page.Values;
-    hadResults = pageResults.Any();
+    moreResultsAvailable = pageResults.Any() && continuationToken != null;
 
     // Print out the results for this page.
     foreach (TableEntity result in pageResults)

--- a/sdk/tables/Azure.Data.Tables/tests/samples/Sample4_QueryEntities.cs
+++ b/sdk/tables/Azure.Data.Tables/tests/samples/Sample4_QueryEntities.cs
@@ -112,8 +112,8 @@ namespace Azure.Data.Tables.Samples
             #region Snippet:TablesSample4QueryPagination
 
             string continuationToken = null;
-            bool hadResults = true;
-            while (hadResults)
+            bool moreResultsAvailable = true;
+            while (moreResultsAvailable)
             {
                 Page<TableEntity> page = tableClient
                     .Query<TableEntity>()
@@ -128,7 +128,7 @@ namespace Azure.Data.Tables.Samples
                 continuationToken = page.ContinuationToken;
 
                 IReadOnlyList<TableEntity> pageResults = page.Values;
-                hadResults = pageResults.Any();
+                moreResultsAvailable = pageResults.Any() && continuationToken != null;
 
                 // Print out the results for this page.
                 foreach (TableEntity result in pageResults)

